openlibrary/plugins/openlibrary/api.py

```diff
@@ -658,13 +658,10 @@ class hide_banner(delegate.page):
 
     def POST(self):
         data = json.loads(web.data())
-        user = accounts.get_current_user()
-
-        user.save_preferences({'hidden-banner': data['storage-key']})
 
-        def response(msg, status="success"):
-            return delegate.RawText(
-                json.dumps({status: msg}), content_type="application/json"
-            )
+        # Set truthy cookie that expires in 30 days:
+        web.setcookie(data['cookie-name'], '1', expires=(3600 * 24 * 30))
```

Comment:
I wonder if we want this date to be customizable based on the campaign, e.g. `duration_days=30` by default